### ISSUE TITLE
Model bound incorrect

### ISSUE THREAD

**Issue by paulmelis** [StereoKit/StereoKit#567]
## Description

As mentioned in https://github.com/StereoKit/StereoKit/discussions/564 I'm seeing some strangely sized handle bounds, much larger than need to be based on vertex range. I first thought it might be related to the GLTF reader, but now see the same with a model in .obj format. Looks like the Y range is far off:

![image](https://user-images.githubusercontent.com/1408287/212430184-c6343339-0794-46a9-91b2-30ea31a4a30b.png)

(the model looks weird as I stripped out the normals to keep the file size down, attached below).

## Platform / Environment

Results here with Linux, 0.3.7-preview6, but see the same on UWP.

## Logs or exception details

```
using StereoKit;

class Program
{
	static void Main(string[] args)
	{
		SK.Initialize(new SKSettings{ appName = "Project" });

                Model model = Model.FromFile("doh.obj");
		Log.Info($"c {model.Bounds.center}");
		Log.Info($"d {model.Bounds.dimensions}");

		Pose pose = Pose.Identity;

		SK.Run(() =>
		{            
                        Lines.AddAxis(Pose.Identity, 1.0f);

			UI.HandleBegin("Model Handle", ref pose, model.Bounds);
				model.Draw(Matrix.Identity);
				Mesh.Cube.Draw(Material.UIBox, Matrix.TS(model.Bounds.center, model.Bounds.dimensions));
			UI.HandleEnd();
		});
        }
}
```

This prints
```
[SK info] c [-0.34, -1.77, 0.04]
[SK info] d [1.49, 3.55, 2.57]
```

Checking the actual vertex range with Python:
```
#!/usr/bin/env python
import numpy

coords = [[],[],[]]

with open('doh.obj', 'rt') as f:
    for line in f:
        if not line.startswith('v '):
            continue
        line = line.strip().split(' ')
        for i, v in enumerate(map(float, line[1:])):
            coords[i].append(v)

    for arr in coords:
        arr = numpy.array(arr)
        min = numpy.min(arr)
        max = numpy.max(arr)
        print('%.6f %.6f c %.6f d %.6f' % (min, max, 0.5*(min+max), max-min))

melis@juggle 23:18:~/examples/stereokit/loosebound$ ./bbox.py doh.obj 
-1.086361 0.402941 c -0.341710 d 1.489302
-3.549514 -3.280300 c -3.414907 d 0.269214
-1.243968 1.321764 c 0.038898 d 2.565732
```

So the Y range in SK is indeed not correct. I don't have time to check the SK code right now, but it might be a case of not updating the Y min value when scanning the vertices?

[doh.obj.gz](https://github.com/StereoKit/StereoKit/files/10415805/doh.obj.gz)

**Comment by maluoi** [comment]
Thanks for digging deeper into this! Bounds calculation for the `Mesh` was fine, but the `Model` was trying to encapsulate the origin of the `Model` when adding the node! It's fixed now. While I was at it, I added some `Bounds` functions to the C# API, and ported around some performance improvements that were in some other spots.
 
I also fixed some .obj loading issues that this asset brought up! Faces were referencing non-existent UVs and Normals, and that was occasionally crashing things. I fixed that as well, and now auto-calculate normals if none were found :) Looks a bit better!
![image](https://user-images.githubusercontent.com/8484884/212445666-249a6ae9-d5e9-44b7-9c50-597c4bfee338.png)
